numba/cuda/cudadrv/devicearray.py

```diff
@@ -146,11 +146,26 @@ def copy_to_device(self, ary, stream=0):
         if ary.size == 0:
             # Nothing to do
             return
+        sentry_contiguous(self)
         stream = self._default_stream(stream)
         if _driver.is_device_memory(ary):
+            if self.ndim > 1 and self.flags['C_CONTIGUOUS'] != ary.flags['C_CONTIGUOUS']:
+                raise ValueError("Can't copy %s-contiguous array to a %s-contiguous array" % (
+                    'C' if ary.flags['C_CONTIGUOUS'] else 'F',
+                    'C' if self.flags['C_CONTIGUOUS'] else 'F',
+                ))
+
             sz = min(self.alloc_size, ary.alloc_size)
             _driver.device_to_device(self, ary, sz, stream=stream)
         else:
+            # ensure same contiguous-nous
+            if self.ndim > 1:
```

Comment:
Same as above.  

An example that will break this is:

```python
a_c = np.arange(10)
d = cuda.to_device(a_c)
arr = np.arange(20)[::2]
d.copy_to_device(arr)
np.testing.assert_array_equal(d.copy_to_host(), arr)
```

Reply:
I've added this example to the test cases